src/main.cpp

```diff
@@ -1,13 +1,43 @@
-#pragma once
-
 #include <iostream>
 #include <glog/logging.h>
 #include <SDL.h>
+//#include <cnpy.h>
 #include <exception>
+#include <regex>
 #include "src/game.h"
 
 #undef main //SDL defines main and it needs to be undefined
 
+static const std::string DEFAULT_ROOM_CONTENT_PATH = "valid_images.npy";
+
+std::shared_ptr<RoomContentProvider> create_room_content_provider(int argc, const char* argv[]) {
+    std::string room_content = "none";
+    std::string room_content_path = DEFAULT_ROOM_CONTENT_PATH;
+
+    std::regex contains_static_regex("static");
+    std::regex contains_generate_regex("generate");
+    std::regex room_content_regex("room_content=(none)|(static)|(generate)");
+    std::regex room_content_path_regex("room_content_path=.*.npy");
+
+    for(int i = 1; i < argc; ++i) {
+        if(std::regex_search(argv[i], room_content_regex)) {
+            if(std::regex_search(argv[i], contains_static_regex)) {
+                room_content = "static";
+            } else if(std::regex_search(argv[i], contains_generate_regex)) {
```

Comment:
I think it would be much more readable if you didn't use `contains_generate_regex` nor `contains_static_regex` variables and just their values. They are not reused